### ISSUE TITLE
Running on Alfred 4.0

### ISSUE THREAD

**Issue by r3ver53r** [BenjaminOddou/alfred-dashlane#4]
Hello,

As I understand this requires Alfred 5 to run. Is there a workaround/code-tweaking to run it on alfred 4? Any pointers will be appreciated.

**Comment by BenjaminOddou** [comment]
Hello @r3ver53r,

The first thing you'll need to change to make it work on Alfred 4 is the way scripts access workflow variables. With the release of Alfred 5 came the [configuration panel](https://www.alfredapp.com/help/workflows/workflow-configuration/). I think you can make it work by manually creating the following environment variables [here](https://www.alfredapp.com/help/workflows/advanced/variables/#environment):

- `trigger1` => `pdash`
- `trigger2` => `odash`
- `user_mail` => `youremail@something.com`
- `incognito_mode` => `1`
- `cache_bool` => `1`
- `sound` => `Submarine`

Some of these variables are accessed through the [utils.py](https://github.com/BenjaminOddou/alfred-dashlane/blob/main/src/utils.py) file to be used in the workflow. Tell me if this works, otherwise, I'll try to further assist.

**Comment by r3ver53r** [comment]
Hey @BenjaminOddou - Thanks for looking into this. Unfortunately, this dashlane workflow is not even showing the option to edit env variables, unlike other workflows. As if it's not even getting recognised. Attaching screenshot for ref.

![Screenshot 2024-05-25 at 8 21 05 PM](https://github.com/BenjaminOddou/alfred-dashlane/assets/8873311/193f2431-a5d2-46da-9af5-902233f19c9a)
![Screenshot 2024-05-25 at 8 21 27 PM](https://github.com/BenjaminOddou/alfred-dashlane/assets/8873311/12c92c1a-477f-4082-9ee3-fdb6ed18a54e)

**Comment by BenjaminOddou** [comment]
Hello @r3ver53r, could you try to use this [modified workflow](https://mega.nz/file/OgwmTQDT#y82ZjSpVT7tSAsQlnZDsZHLi2eXMzT00EG2VAx-fVd8) ?

**Comment by r3ver53r** [comment]
Wao, that got imported. Thanks @BenjaminOddou ! :D 
Looks like getting stuck with another small bug, it is not able to detect dashlane cli. (screenshots attached)

![dashlane-cli-error](https://github.com/BenjaminOddou/alfred-dashlane/assets/8873311/b6bf20e2-ee64-4771-b6f5-dd2c2e8af12d)
![Screenshot 2024-05-26 at 9 15 01 AM](https://github.com/BenjaminOddou/alfred-dashlane/assets/8873311/5e995622-bd11-4ce7-bac4-f05a99708204)

**Comment by BenjaminOddou** [comment]
Hello @r3ver53r, could you check that the environment variable alfred_workflow_cache corresponds to : `~/Library/Caches/com.runningwithcrayons.Alfred/Workflow Data/com.benjamino.dashlane` ? Replace `~` by `/Users/yourusername`.

**Comment by r3ver53r** [comment]
Hi @BenjaminOddou  - Yes. This is my ENV: `/Users/<MyUserName>/Library/Caches/com.runningwithcrayons.Alfred/Workflow Data/com.benjamino.dashlane`

**Comment by BenjaminOddou** [comment]
@r3ver53r, could you try to comment these lines from [entry.py](https://github.com/BenjaminOddou/alfred-dashlane/blob/main/src/entry.py) :

1. [Line 128](https://github.com/BenjaminOddou/alfred-dashlane/blob/18fd7a12f3d33838b8fc06376f8cd6949cefabed/src/entry.py#L128C1-L128C9)
2. [Lines 176-189](https://github.com/BenjaminOddou/alfred-dashlane/blob/18fd7a12f3d33838b8fc06376f8cd6949cefabed/src/entry.py#L176C1-L189C10)

and check the Alfred Logs ?

**Comment by r3ver53r** [comment]
I guess I figured out the bug (though, not root cause)...It looks like the which() function in pexpect utils (https://github.com/BenjaminOddou/alfred-dashlane/blob/18fd7a12f3d33838b8fc06376f8cd6949cefabed/src/lib/pexpect/utils.py#L48) is not able to resolve my local dcli path

I did a patchy work-around by changing all 'dcli' references to '/opt/homebrew/bin/dcli' (exact path) and it seems to working. Thanks for your support so far :)

Happy to dig this further on why it's not able to find the dcli path in you plan to support (or add workarounds) Alfred 4 for other users. 

Big thanks!
